packages/notifier/src/publish-kit.js

```diff
@@ -180,12 +186,209 @@ export const makePublishKit = () => {
+        new Error('Cannot read past end of iteration.'),
+      ),
+      tailR: undefined,
+    };
+  } else if (status === 'finished') {
+    // XXX Should we be harden()ing rejection-reason errors?
+    const tailP = makeQuietRejection(
+      new Error('Cannot read past end of iteration.'),
+    );
+    newData = {
+      currentP: HandledPromise.resolve(
+        harden({
+          head: { value: state.value, done: true },
+          publishCount,
+          tail: tailP,
+        }),
+      ),
+      tailP,
+      tailR: undefined,
+    };
+  } else if (status === 'live') {
+    const { promise: tailP, resolve: tailR } = makePromiseKit();
+    newData = {
+      currentP: state.hasValue
+        ? HandledPromise.resolve(
+            harden({
+              head: { value: state.value, done: false },
+              publishCount,
+              tail: tailP,
+            }),
+          )
+        : tailP,
+      tailP,
+      tailR,
+    };
+  } else {
+    assert.fail(X`Invalid durable promise kit status: ${q(status)}`);
+  }
+  durablePublishKitEphemeralData.set(state, harden(newData));
+  return newData;
+};
+
+/**
+ * Extends the sequence of results.
+ *
+ * @param {DurablePublishKitState} state
+ * @param {boolean} done
+ * @param {any} value
+ * @param {any} [rejection]
+ */
+const advanceDurablePublishKit = (state, done, value, rejection) => {
+  const { valueDurability, publishCount: oldPublishCount, status } = state;
+  if (status !== 'live') {
+    throw new Error('Cannot update state after termination.');
+  }
+  if (done || valueDurability === 'mandatory') {
+    assertPassable(value);
+    assertPassable(rejection);
+  }
+  const { tailP: currentP, tailR: resolveCurrent } =
+    provideDurablePublishKitEphemeralData(state);
+
+  const publishCount = oldPublishCount + 1n;
+  state.publishCount = publishCount;
+  let tailP;
+  let tailR;
+
+  if (done) {
+    state.status = rejection ? 'failed' : 'finished';
+    tailP = makeQuietRejection(new Error('Cannot read past end of iteration.'));
+    tailR = undefined;
+  } else {
+    ({ promise: tailP, resolve: tailR } = makePromiseKit());
+  }
+  durablePublishKitEphemeralData.set(state, harden({ currentP, tailP, tailR }));
+
+  if (rejection) {
+    state.value = rejection;
+    state.hasValue = true;
+    resolveCurrent(rejection);
+  } else {
+    state.hasValue = false;
+    try {
+      if (done || valueDurability !== 'ignored') {
+        assertPassable(value);
+        state.value = value;
+        state.hasValue = true;
+      }
+    } catch (err) {
+      if (done || valueDurability === 'mandatory') {
+        throw err;
+      }
+    }
+    resolveCurrent(
+      harden({
+        head: { value, done },
+        publishCount,
+        tail: tailP,
+      }),
+    );
+  }
+};
+
+/**
+ * @param {import('../../vat-data/src/types.js').Baggage} baggage
+ * @param {string} kindName
+ */
+export const vivifyDurablePublishKitMaker = (baggage, kindName) => {
```

Comment:
I prefer `vivifyDurablePublishKit`.   With `vivifyFooMaker` I have trouble telling whether it's vivifying a maker, or a maker of vivifyFoos.  I've gotten used to the idea that `vivifyFoo` and `vivifyFooKit` return makers, and `vivifyFooInstance` returns an instance.

Reply:
So concretely, are you asking for something that supports the following?
```js
const makeDurablePublishKit = vivifyDurablePublishKit(baggage, 'publish kit');
const { publisher: publisher1, subscriber: subscriber1 } = makeDurablePublishKit();
const { publisher: publisher2, subscriber: subscriber2 } = makeDurablePublishKit();
```

Reply:
Yeah. That's what I was looking for.

Reply:
Done.